core/lib/types/Cargo.toml

```diff
@@ -32,6 +32,8 @@ num_enum = "0.6"
 hex = "0.4"
 prost = "0.12.1"
 
+sqlx = "0.7.3"
```

Comment:
This is **very** dangerous. One of the main purposes of DAL crate was not to give direct access to SQLx to any dependent crates and make sure that all of our DB-related stuff is well encapsulated.

At the same time, `zksync_types` is a crate where people move stuff when they need to share something but don't know where to put it. Which means that virtually everything in zkSync stack depends on `zksync_types`. This crate is already pretty toxic in APIs it exposes, and it makes it even more toxic: people use what they can use and in a rush they may do hacks that are hard to fix later.

AFAICS you only need sqlx to expose `chrono` types, which are [just reexports](https://github.com/launchbadge/sqlx/blob/main/sqlx-core/src/types/mod.rs#L40). Instead you can depend on [the same version of chrono](https://github.com/launchbadge/sqlx/blob/v0.7.3/sqlx-core/Cargo.toml#L59) and simply use these types directly. Which [seemingly we already do](https://github.com/matter-labs/zksync-era/blob/cadffe6410385bd6fb62c5a7646ae2b8272d56e2/core/lib/types/Cargo.toml#L22) so probably you don't need this dependency at all.

Reply:
thanks, nice suggestion